mongo/client.go

```diff
@@ -705,6 +702,11 @@ func (c *Client) configure(opts *options.ClientOptions) error {
 		c.deployment = opts.Deployment
 	}
 
+	topologyOpts = append(topologyOpts, topology.WithServerOptions(
```

Comment:
Any reason to move this below the `c.deployment` assignment?

Reply:
Yes. The `if opts.Deployment != nil` block checks the length of `topologyOpts` to see if there are any non-default topology options specified along with a custom Deployment. Updating the append logic to append the server options to `topologyOpts` instead of to `c.topologyOptions` means the length of `topologyOpts` is greater than the expected length, resulting in `configure` returning error
> cannot specify topology or server options with a deployment

Thinking about this more, the current "no topology or server options with custom deployments" check is extremely brittle, so I may try to move that check to the beginning of `configure` so the check doesn't depend on seemingly unrelated code in the rest of the function. If that's difficult, I'll add a comment describing the reason for the location of the "custom topology options" block.

Reply:
I decided to keep the existing `append` operation order and increase the expected "defaults length" by one. I'll submit a separate PR with the proposed "no topology or server options with custom deployments" check improvement.